### PR TITLE
Controller documentation

### PR DESCRIPTION
**Pull request by iandareid**
Here is all of the documentation for the website on the controller. It will include in the end all of the relevant information on each of the control loops, a guide on tuning the controller and short summary guide to editing and modifying the controller.

### CODE REVIEW

**Comment by iandareid** [comment]
One question I have is whether I should add a diagram for the inner loop and outer loop for each of the controllers. It would make things more clear, but perhaps all I have to do is highlight an example loop when briefly discussing successive loop closure as a whole.

File: docs/developer-guide/rosplane/controller-software-architecture.md
Comment (JMoore5353) [inline docs/developer-guide/rosplane/controller-software-architecture.md:1]: Missing a word, I think, on line 7. Should it say "...inherit from a particular class *and* make the modifications..."?

**Comment by JMoore5353** [comment]
> One question I have is whether I should add a diagram for the inner loop and outer loop for each of the controllers. It would make things more clear, but perhaps all I have to do is highlight an example loop when briefly discussing successive loop closure as a whole.

I think you could also just draw a box behind the inner loop on each of the diagrams, and label the box "Inner Loop". That might be more clear and take up less space than making inner and outer loop diagrams for everything. Thoughts?

**Comment by iandareid** [comment]
I have finished all of the controller documentation. It looks like other documentation has been put onto this branch as well, I will leave it to you @JMoore5353 to decide when to merge this/make this a full PR rather than a draft.

**Comment by iandareid** [comment]
@JMoore5353 Could we rename the directory "Path Planning" to something like "Navigation"? Path planning referencing both a specific module and the entire path part of the autopilot I think could be confusing.

**Comment by JMoore5353** [comment]
> I have finished all of the controller documentation. It looks like other documentation has been put onto this branch as well, I will leave it to you @JMoore5353 to decide when to merge this/make this a full PR rather than a draft.

Sorry about that. I should have made a new branch. I wasn't thinking about that!

**Review by JMoore5353 — APPROVED** [review]

**Pull request merged**